### DISCUSSION TITLE
Export to LibreOffice - Headings and Body Text

### DISCUSSION THREAD

**Discussion by bnpwalter** [freeplane/freeplane#1865] [Help]
When I export the Outline View of Freeplane to an ODT file, node levels translate into Headings in LibreOffice Writer. 
Is there a way to create a Level so that when one exports into LO writer that level is **not a Heading Style** but rather the **Body Text style**?

**Comment by Olifair** [comment]
After what I have seen (but I'm by no means a Freeplane programmer, just a user), the LO styles for the output are hardcoded into Freeplane. 
It would be a really good improvement if one could choose the way Freeplane translates node hierarchy into different paragraphe styles
Actually in the moment only the central node (or the first node if exporting branches) will be in a header style, the remaining nodes, indiscriminately from their node hierarchy are just normal body text, and even notes are just in between,. 
Even a simple TAB-multiple by node depth would be more usefull.

**Comment by bnpwalter** [comment]
My experience is that not only the central node but each sub-level of nodes are exported as headings.  See attached.  If you look at the LibreOffice Navigator on the left side panel you will see that the **third paragraph** starting with the words "You can make up the deficiency..." is actually **Heading 3.**
The fourth paragraph starting with "This is a node detail: Let's see if it prints" is a **node detail** and is not a heading but **Default Paragraph Style.**

![FP EXPORT TO LIBREOFFICE - LEVELS-HEADINGS](https://github.com/freeplane/freeplane/assets/50636483/57a41deb-4607-4b05-9ade-884b92ac67dd)

If one only wants to convert a single sub-heading (or sub-level) into the body text style within LibreOffice writer it's not too difficult. One simply uses the **Find and Replace** function to change that sub-heading style (paragraph style) to Body Text Style.  This becomes more cumbersome should you want to change numerous sub-heading styles to body text.

**Node details**, however, do not import into Libreoffice as Headings.  They import in LO as the **default text style**.  See above.  This is encouraging.  It implies that there may be a way to program FP nodes in a similar fashion to node details and perhaps allow one to choose a style other than a heading style.   (I'm just a user as well)

That said, I agree with you.  It would be a welcomed improvement if one could choose how FP exports levels.   How does one make that feature request on Github?

**Reply by Olifair** [reply]
@bnpwalter "My experience is that not only the central node but each sub-level of nodes are exported as headings. " Yes, that's what I thought I remembered from past exports, however when I did several trials from several different maps yesterday, using Freeplane 1.11.13 and Libreoffice 24.2.2.2 
![Freeplane Export test map](https://github.com/freeplane/freeplane/assets/6613705/4f1e5756-9fe7-4c40-8f34-4679bb12c587)

![LibreOffice Exported test map](https://github.com/freeplane/freeplane/assets/6613705/05b8c0bd-0084-458a-83f5-3d02a3b31f21)

so I don't know if this is a version difference, or if certain styles in Freeplane translate into certain styles in LibreOffice ?

**Comment by bnpwalter** [comment]
I'm working on Windows 10 with FP version 1.11.14.  Think I figured this out. 
If you go to View>View Settings>(check) Outline view.  In **Node Styles:** Select **Automatic level styles: For all Nodes.**   After creating your nodes at various levels ... ie Level 1, Level 2, Level 3 etc.  then select the level you want to export as Body Text  and change the Automatic Level Styles: to  **"for non leaf nodes"**  This will set the **Node Style** for that Level globally to **Default.**  Default exports as Body Text in LO.
See attached. 
![_Change a Node Level in FP to export as Body Text](https://github.com/freeplane/freeplane/assets/50636483/4475e223-21bf-48df-b088-436950a1a9d2)
![_For non leaf nodes exported as Body Text in LO](https://github.com/freeplane/freeplane/assets/50636483/24875826-fa08-4059-af83-784c4716a196)

In the example above, I've changed **Level 4**  ( shown as AAAA) to Automatic level Styles: **for non leaf nodes.**  Changing AAAA to for non leaf nodes automatically changed Level 4 (BBBB) to the **Default** style. 

Hope this makes sense.

**Comment by Olifair** [comment]
@bnpwalter thanks for this idea, never had a look into the outline view and thought that this would impact the export. I'll have to play with this
